packages/next-server/lib/router/router.ts

```diff
@@ -21,6 +21,8 @@ function toRoute(path: string): string {
   return path.replace(/\/$/, '') || '/'
 }
 
+export type RouterUrl = string | UrlObject | Url
```

Comment:
`Url` appears to be a sub-set of `UrlObject` type: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d9f74cb42cbd76a4a8006424f5da00767852552a/types/node/url.d.ts#L17-L27

The difference brings down to `ParsedUrlQuery` vs `ParsedUrlQueryInput`, where the latter is less strict: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/132720a17e15cdfcffade54dd4a23a21c1e16831/types/node/querystring.d.ts#L11-L18

Overall, it seems like `string | UrlObject` is sufficient, unless I'm missing something 🤔